MoreLinq.Test/CountMethodsTestGenerator.cs

```diff
@@ -0,0 +1,42 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Avi Levin. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System;
+	using System.Collections.Generic;
+	using System.Text;
+
+	public static class CountMethodsTestGenerator
+    {
+        public static IEnumerable<TestCaseData> GetTestCases
+        (
+            IEnumerable<(int size, int comparedTo)> counts,
+            Func<int, int, bool> expectedResultCalculator
+        )
+		{
+            return (
+				from count in counts
+                from type in new[] { SourceKind.Sequence, SourceKind.BreakingCollection, SourceKind.BreakingReadOnlyCollection }
+                select new TestCaseData(type, count.size, count.comparedTo)
+					.Returns(expectedResultCalculator(count.size, count.comparedTo))
+					.SetName($"{{m}}({type}[{count.size}], {count.comparedTo}")
```

Comment:
To answer the question (and get back to whether it is worth it later):

> {m} The method name of the test or empty if there is no method. The name includes any type arguments, enclosed in angle braces and separated by commas. ([documentation](https://github.com/nunit/docs/wiki/Generation-of-Test-Names-Spec))

This allows the method to generate appropriately named tests (e.g. "AtMost...") without passing it in explicitly.

Reply:
No I get that but it's going to get formatted literally as `{m}` since `{` and `}` have been escaped. It's never going to get replaced or substituted.

Reply:
Once the string interpolation is resolved, it is passed to the function as a standard `{m}`.

Reply:
Ah right, that's to do with [template-based test naming that's part of NUnit](https://github.com/nunit/docs/wiki/Template-Based-Test-Naming)! All good now.